File: src/main/kotlin/ui/ChooseSheetDialog.java
package ui;

import model.HiidoStaticSheet;

import javax.swing.*;
import java.awt.event.*;
import java.util.HashMap;
import java.util.Map;

public class ChooseSheetDialog extends JDialog {
    private JPanel contentPane;
    private JButton buttonOK;
    private JButton buttonCancel;
    private Map<String, HiidoStaticSheet> mData = new HashMap<>(0);

    public ChooseSheetDialog() {
        setContentPane(contentPane);
        setModal(true);
        getRootPane().setDefaultButton(buttonOK);

        buttonOK.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                onOK();
            }
        });

        buttonCancel.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                onCancel();
            }
        });

        // call onCancel() when cross is clicked
        setDefaultCloseOperation(DO_NOTHING_ON_CLOSE);
        addWindowListener(new WindowAdapter() {
            public void windowClosing(WindowEvent e) {
                onCancel();
            }
        });

        // call onCancel() on ESCAPE
        contentPane.registerKeyboardAction(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                onCancel();
            }
        }, KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0), JComponent.WHEN_ANCESTOR_OF_FOCUSED_COMPONENT);
    }
    private void updateUI() {

    }


    public void setData(Map<String, HiidoStaticSheet> data) {
        mData = data;
        updateUI();
    }

    private void onOK() {
        // add your code here
        dispose();
    }

    private void onCancel() {
        // add your code here if necessary
        dispose();
    }

    public static void main(String[] args) {
        ChooseSheetDialog dialog = new ChooseSheetDialog();
        dialog.pack();
        dialog.setVisible(true);
        System.exit(0);
    }
}

File: src/main/kotlin/ui/ChooseSheetDialog.kt
package ui

import model.HiidoStaticSheet
import utils.OnConfirmListener
import utils.centerDialog
import java.awt.Dimension
import java.awt.Insets
import java.awt.event.KeyEvent
import java.awt.event.WindowAdapter
import java.awt.event.WindowEvent
import javax.swing.*

/**
 * 读取Excel files 后选择目标sheets
 * */
class ChooseSheetDialog(private var mData: HashMap<String, HiidoStaticSheet>?, var listener: OnConfirmListener<HiidoStaticSheet?>?) : JDialog() {
    private var contentPane: JPanel? = null
    private var buttonOK: JButton? = null
    private var buttonCancel: JButton? = null
    private var headerView: JPanel? = null
    private var sheetsList: JPanel? = null
    private var buttonGroup: ButtonGroup? = null


    init {
        centerDialog(400, 300)
        setContentPane(contentPane)
        buttonGroup = ButtonGroup()
        isModal = true
        getRootPane().defaultButton = buttonOK
        buttonOK!!.addActionListener { onOK() }
        buttonCancel!!.addActionListener { onCancel() }
        // call onCancel() when cross is clicked
        defaultCloseOperation = WindowConstants.DO_NOTHING_ON_CLOSE
        addWindowListener(object : WindowAdapter() {
            override fun windowClosing(e: WindowEvent?) {
                onCancel()
            }
        })
        contentPane!!.registerKeyboardAction(
            { onCancel() },
            KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0),
            JComponent.WHEN_ANCESTOR_OF_FOCUSED_COMPONENT
        )
        addHeader()
        addSheetsList()
    }


    private fun addHeader() {
        headerView!!.layout = BoxLayout(headerView, BoxLayout.X_AXIS)
        val label1 = JLabel("选择表格")
        label1.preferredSize = Dimension(100, 25)
        headerView!!.add(label1)
    }

    private fun addSheetsList() {
        sheetsList!!.layout = BoxLayout(sheetsList, BoxLayout.Y_AXIS)
        mData?.forEach { sheetsName, _ ->
            val radioButton = JRadioButton(sheetsName)
            radioButton.margin = Insets(20, 20, 20, 20)
            sheetsList?.add(radioButton)
            buttonGroup?.add(radioButton)
        }
    }

    private fun onOK() {
        dispose()
        mData?.let { map ->
            val sheetName = buttonGroup?.elements?.toList()?.filter {
                it.isSelected
            }?.get(0)?.text
            sheetName?.let {
                listener?.onConfirm(map[it])
            }
        }
    }

    private fun onCancel() {
        dispose()
    }
}
